### DISCUSSION TITLE
Is it possible to rotate the NumericUpDown control?

### DISCUSSION THREAD

**Discussion by dknippel** [MahApps/MahApps.Metro#4398] [Q&A] [answered]
Hi @punker76 , When I try to apply a 90 degree rotate transform, the buttons and text are rotated, but not the control itself.

**Comment by dknippel** [comment]
Here is my current XAML that does not seem to rotate the control itself:

```                        
<mah:NumericUpDown Grid.Column="2" Grid.Row="0" Minimum="-180" Maximum="180" 
    Value="{Binding RotateZ}" RenderTransformOrigin="0.5, 0.5">
        <mah:NumericUpDown.RenderTransform>
            <TransformGroup>
                <RotateTransform Angle="90"/>
            </TransformGroup>
        </mah:NumericUpDown.RenderTransform>
</mah:NumericUpDown>
```

**Reply by punker76** [reply]
Hi @dknippel there are dependency properties available to change the alignement of the text and buttons:

**ButtonsAlignment="Left"**

![2023-04-27_14h32_30](https://user-images.githubusercontent.com/658431/234863349-0cc11e72-7537-4b4b-a771-b2578a7d4092.png)

**SwitchUpDownButtons="True"**
![2023-04-27_14h33_01](https://user-images.githubusercontent.com/658431/234863360-64092954-1a75-4f12-b4d7-dc6b049e7393.png)

**TextAlignment="Left"**
![2023-04-27_14h33_52](https://user-images.githubusercontent.com/658431/234863370-cba7fd16-4c46-4eae-9a7c-bed00b4e2eae.png)

**Comment by dknippel** [comment]
Hi @punker76 , thanks so much for your reply!  I'm aware of the dependency properties available to change the alignement of the text and buttons.  Those work great.

However, due to screen real estate limits I am needing to rotate the entire control so that it renders vertically.  The text and buttons do rotate correctly.  Unfortunately though the containing "box" does not seem to rotate and thus I'm not able to place multiple numericupddown controls close enough to each other.  I've attached a screenshot for your reference.  The red box seems to be some kind of container element for the numericupdown control that is not rotating and thus preventing closer arrangement of multiple instances of said control.

Do you know if rendering the **_entire_** numericupdown control vertically is possible?  If so, what would be the correct XAML?

![rotate_numericupdown](https://user-images.githubusercontent.com/35277981/234897897-39598226-77d2-424a-b4f6-a479a05ec614.png)

**Comment by punker76** [comment, marked as answer]
@dknippel Ok, got it, you should use LayoutTransform instead RenderTransform to get this:

![2023-04-27_17h11_51](https://user-images.githubusercontent.com/658431/234907004-3a9d10e4-cc4b-4ebd-b093-6c8d027dfac0.png)

I hope this is what you want.

```xaml
    <Grid HorizontalAlignment="Center" Grid.ShowGridLines="True">
        <Grid.ColumnDefinitions>
            <ColumnDefinition Width="Auto" />
            <ColumnDefinition Width="Auto" />
            <ColumnDefinition Width="Auto" />
        </Grid.ColumnDefinitions>

        <mah:NumericUpDown Grid.Column="0"
                           Grid.Row="0"
                           Minimum="-180"
                           Maximum="180" 
                           Value="100"
                           Margin="4"
                           VerticalAlignment="Stretch"
                           HorizontalAlignment="Stretch">
            <mah:NumericUpDown.LayoutTransform>
                <TransformGroup>
                    <RotateTransform Angle="90"/>
                </TransformGroup>
            </mah:NumericUpDown.LayoutTransform>
        </mah:NumericUpDown>

        <mah:NumericUpDown Grid.Column="1"
                           Grid.Row="0"
                           Minimum="-180"
                           Maximum="180" 
                           Value="100"
                           Margin="4"
                           VerticalAlignment="Stretch"
                           HorizontalAlignment="Stretch">
            <mah:NumericUpDown.LayoutTransform>
                <TransformGroup>
                    <RotateTransform Angle="90"/>
                </TransformGroup>
            </mah:NumericUpDown.LayoutTransform>
        </mah:NumericUpDown>

        <mah:NumericUpDown Grid.Column="2"
                           Grid.Row="0"
                           Minimum="-180"
                           Maximum="180" 
                           Value="100"
                           Margin="4"
                           VerticalAlignment="Stretch"
                           HorizontalAlignment="Stretch">
            <mah:NumericUpDown.LayoutTransform>
                <TransformGroup>
                    <RotateTransform Angle="90"/>
                </TransformGroup>
            </mah:NumericUpDown.LayoutTransform>
        </mah:NumericUpDown>
    </Grid>
```

**Reply by dknippel** [reply]
Haha, yes that worked!!!  Also, now I understand a bit more about `RenderTransform` versus `LayoutTransform`. Thanks so much @punker76  :-D
